src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -334,30 +340,74 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
     }
 
     @Override
-    public Map<String, Long> materializePositions(final String topicId, final SubscriptionBase.InitialPosition position)
+    public Map<String, Long> materializePositions(final EventType eventType, final Subscription subscription)
```

Comment:
No, I think it works only in opposite direction. Service layer should not know anything about specific repository implementation. But repository should use domain objects, in other case it's not clear how we can use repositories at all.